### DISCUSSION TITLE
APIMAN 3.1 - Communication problem between the GATEWAY VERT.X and KEYCLOAK V20

### DISCUSSION THREAD

**Discussion by tsilaRaf** [apiman/apiman#2524] [Help]
Good morning,

We want to install version 3.1 of the APIMAN Gateway in vertx.
We use Keycloak Quarkus (version 20) as authentication server and elasticsearch version 8.6 as database.
We are having a problem because we cannot reach the keycloak url and we have the following error message:

thp-api-gw-vertx | [ERROR] 2023-05-05 10:44:00.634 [vert.x-eventloop-thread-1] KeycloakOAuthFactory - Failure during discovery
thp-api-gw-vertx | io.vertx.core.impl.NoStackTraceThrowable: Not Found: {"error":"RESTEASY003210: Could not find resource for full path: https://{alias dns}**/null**/realms/apiman/.well-known/openid -configuration"}

**We can observe:**
-The gateway adds /null in after https://{alias dns}/

**Track:**
-Keycloak(version 15.1) => url path: https://{alias dns}/auth
-Keycloak(version 20) => url path: https://{alias dns}/

We tested with another Gateway APIMAN (version 2.3) wildfly , on the same version of the keycloak (version 20) and we don't have the problem.

**Below is an excerpt from our Gateway configuration:**

"auth": {
     "type": "keycloak",
     "config": {
       "flowType": "PASSWORD",
       "requiredRole": "realm:apipublisher",
"realm": "apiman",
"realm-public-key": "${KEYCLOAK_PUBLIC_KEY}",
       "auth-server-url": "https://${alias dns}",
       "ssl-required": "none",
       "resource": "apiman-gateway-api",
"credentials": {
         "secret": "${KEYCLOAK_GW_SECRET}"
       }
     }
}

**List of our software components:**
- APIMAN (version 3.1.0: Gateway vert.x + Manager wildfly)
- JAVA JEE application server: WILDFLY (version 23.0.2)
- ELASTICSEARCH (version 8.6)
- KIBANA (version 8.6)
- Postgres (version 15)
- KEYCLOAK (version 20.0.3)
- VAULT (version 1.11.2)

Could you help us solve our problem?

THANKS,
Cordially,

**Comment by msavy** [comment]
This will be fixed in 3.2, thanks.

**Reply by tsilaRaf** [reply]
Thank you for your answer,

Can you tell me approximately when version 3.2 will be available?

thank you,

**Reply by msavy** [reply]
Sorry, can't provide any updates at the moment for upstream community release. I'm still working on some stuff for paying customers.

**Reply by msavy** [reply]
I should add, if you're a support customer, feel free to reach out via email if you need something backporting/fixing urgently.
